feature(admin): 관리자 기능 페이지 UI 구현

sinyeowon opened the pull request:
## ✨ 요약

> 스터디 관리자를 위한 종합 관리 페이지를 구현했습니다. 스터디원 관리, 신청자 승인/거절, 스터디 정보 수정 기능을 탭 인터페이스로 제공합니다.

## 🔗 작업 내용

- 관리자 페이지 기본 구조 및 탭 네비게이션 구현
- 스터디원 관리 기능 (멤버 목록 조회, 탈퇴 처리)
- 신청자 관리 기능 (승인/거절 처리)
- 스터디 정보 수정 기능 (스터디 생성 페이지와 동일한 폼 구조)
- 사이드바 고정 및 스크롤 최적화

## 💻 상세 구현 내용

**컴포넌트 구조:**
- `AdminPage`: 메인 관리자 페이지 컴포넌트, 탭 상태 관리
- `AdminTabs`: 탭 네비게이션 컴포넌트 (스터디원 관리/신청자 관리/스터디 관리)
- `MemberManagement`: 스터디원 목록 및 탈퇴 기능
- `ApplicantManagement`: 신청자 목록 및 승인/거절 기능
- `StudyInfoManagement`: 스터디 정보 수정 폼 (React Hook Form 사용)

**기술적 특징:**
- TypeScript로 타입 안전성 확보
- React Hook Form을 활용한 폼 상태 관리
- 기존 RegionSelectModal 컴포넌트 재사용
- 목업 데이터를 통한 UI 테스트 가능
- 반응형 레이아웃 및 일관된 디자인 시스템 적용

**레이아웃 개선:**
- 사이드바를 fixed로 고정하여 스크롤 시 콘텐츠 가림 문제 해결
- 헤더와 탭 네비게이션 고정으로 스크롤 중에도 접근 가능
- 문서 관리 페이지와 동일한 헤더 레이아웃 적용

## 🔗 참고 사항

- 현재는 목업 데이터를 사용하며, 실제 API 연동은 추후 작업 예정
- 스터디 생성 페이지의 폼 구조를 재사용하여 일관성 유지
- 기존 디자인 시스템 및 컴포넌트를 최대한 활용

## 📸 스크린샷 (Screenshots)


## 🔗 관련 이슈

- Close #10 

___
### 😊 리뷰 규칙을 지킵시다
코드 리뷰는 `Pn`룰에 따라 작성하기.   
Reviewer가 피드백을 남길 때 Assignee에게 얼마나 해당 피드백에 대해 강조하고 싶은 지 표현하기 위한 규칙입니다.
- `P1` : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
- `P2` : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
- `P3` : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

sinyeowon committed:
feat(admin): 관리자 페이지 기본 타입 정의 및 목업 데이터 수정

- StudyMember, StudyApplicant, StudyInfo 인터페이스 추가
- AdminTabType 타입 정의
- MOCK_STUDY_INFO에 schedule, region, conditions 필드 추가

sinyeowon committed:
feat(admin): AdminPage 메인 컴포넌트 기본 구조 구현

sinyeowon committed:
feat(admin): AdminTabs 컴포넌트 구현

sinyeowon committed:
feat(admin): MemberManagement 컴포넌트 구현

sinyeowon committed:
feat(admin): ApplicantManagement 컴포넌트 구현

sinyeowon committed:
feat(admin): StudyInfoManagement 컴포넌트 구현

sinyeowon committed:
feat(admin): 관리자 페이지 라우팅 설정

sinyeowon committed:
feat(admin): 관리자 페이지 UI/UX 및 레이아웃 개선

- 제목을 '관리자 기능'으로 변경하고 문서 관리 페이지와 레이아웃 통일
- 신청자 관리 탭의 아이콘 버튼을 '수락'/'거절' 텍스트 버튼으로 변경
- 스터디 관리 탭에 이미지 업로드 기능 추가
- 사이드바 고정으로 스크롤 시 콘텐츠 가림 문제 해결
- 헤더와 탭 네비게이션 고정으로 스크롤 중에도 접근 가능

jijun0129 commented on src/pages/(study)/admin/AdminPage.tsx:
```diff
@@ -0,0 +1,52 @@
+  const [activeTab, setActiveTab] = useState<AdminTabType>('members');
+
+  const renderContent = () => {
+    switch (activeTab) {
+      case 'members':
+        return <MemberManagement />;
+      case 'applicants':
+        return <ApplicantManagement />;
+      case 'study-info':
+        return <StudyInfoManagement />;
+      default:
+        return <MemberManagement />;
+    }
+  };
+
+  return (
+    <div className='h-full flex flex-col bg-background'>
+      {/* 헤더 - 고정 */}
+      <div className='flex items-center justify-between px-6 py-6 border-b border-border bg-background flex-shrink-0'>
+        <div>
+          <h1 className='text-2xl font-bold text-primary'>관리자 기능</h1>
+        </div>
+      </div>
+
+      {/* 탭 네비게이션 - 고정 */}
+      <div className='px-6 pt-6 bg-background flex-shrink-0'>
+        <AdminTabs activeTab={activeTab} onTabChange={setActiveTab} />
+      </div>
+
+      {/* 스크롤 가능한 콘텐츠 영역 */}
+      <div className='flex-1 px-6 pb-6 overflow-auto'>
+        <div className='bg-card rounded-lg border border-border p-6'>
+          {renderContent()}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default AdminPage;
```
현재는 탭 상태를 useState로만 관리하고 있어서 새로고침 시 항상 기본 탭으로 초기화되고 있어요
탭이 하위 페이지의 역할을 하고 있으니 `/admin/members`, `/admin/applicants`, `/admin/study-info` 처럼 하위 페이지 url로 구성하면 새로고침해도 탭이 유지되고 네비게이션도 가능할 것 같아요!

sinyeowon replied:
피드백 감사합니다!
탭 상태를 URL 기반으로 관리하도록 개선하였습니다.

jijun0129 reviewed (commented):
수고하셨습니다!

말씀 드린 부분만 한 번 확인해주세요

sinyeowon committed:
feat: 관리자 페이지 URL 기반 탭 관리

- 탭 상태를 useState에서 URL 기반으로 변경
- /admin/members, /admin/applicants, /admin/study-info 라우트 추가
- 새로고침해도 탭 상태 유지 및 브라우저 네비게이션 지원

jijun0129 commented on src/pages/(study)/admin/AdminPage.tsx:
```diff
@@ -0,0 +1,63 @@
+      </div>
+
+      {/* 스크롤 가능한 콘텐츠 영역 */}
+      <div className='flex-1 px-6 pb-6 overflow-auto'>
+        <div className='bg-card rounded-lg border border-border p-6'>
+          {renderContent()}
+        </div>
+      </div>
+    </div>
```
```suggestion
      <div className='flex-1 px-6 pb-6 overflow-auto'>
        <div className='bg-card rounded-lg border border-border p-6'>
          <Outlet />
        </div>
      </div>
    </div>
```

jijun0129 commented on src/routes/Router.tsx:
```diff
@@ -112,7 +113,18 @@ const router = createBrowserRouter([
+              />
+            ),
+          },
+          { path: ROUTES.STUDY.ADMIN.MEMBERS, element: <AdminPage /> },
+          { path: ROUTES.STUDY.ADMIN.APPLICANTS, element: <AdminPage /> },
+          { path: ROUTES.STUDY.ADMIN.STUDY_INFO, element: <AdminPage /> },
```
```suggestion
          { path: ROUTES.STUDY.ADMIN.MEMBERS, element: <MemberManagement /> },
          { path: ROUTES.STUDY.ADMIN.APPLICANTS, element: <ApplicantManagement /> },
          { path: ROUTES.STUDY.ADMIN.STUDY_INFO, element: <StudyInfoManagement /> },
```

sinyeowon replied:
Outlet을 사용해서 하위 페이지 컴포넌트를 보여주는 방식으로 구현하면 더 명확하고 간단하게 페이지를 구성할 수 있을 것 같아요.

말씀해주신대로 Outlet을 사용하는 방식으로 개선 완료했습니다!

jijun0129 reviewed (commented):
Outlet을 사용해서 하위 페이지 컴포넌트를 보여주는 방식으로 구현하면 더 명확하고 간단하게 페이지를 구성할 수 있을 것 같아요.

sinyeowon committed:
refactor: 관리자 페이지 Outlet 기반 구조로 개선

- AdminPage에서 Outlet 사용으로 하위 컴포넌트 렌더링
- 중첩 라우트로 각 탭을 독립적인 페이지로 분리
- 상대 경로 사용으로 라우트 구조 정상화
- ESLint 오류 수정

jijun0129 reviewed (approved):
잘 동작하는 것 확인했습니다. 수고하셨습니다!

머지하겠습니다.
